### PR TITLE
fix: unique id in SfIcon for linearGradient

### LINKED ISSUE #2134: [BUG] Bad value 1 for attribute id on element linearGradient
**Describe the bug**
Bad value 1 for attribute id on element linearGradient

**How to reproduce**
1. Go to SfAccordion
2. See HTML
3. See wrong id for linearGradient

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes #2134 

# Scope of work

# Screenshots of visual changes

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

**Review by AdamPawlinski — APPROVED** [review]

**Review by AdamPawlinski — COMMENTED** [review]
There is a problem when coverege prop changes, icon disappears
icon_coverage
.
Also when you are changing the prop value, the id of linear gradient also changes. I'm not sure if could be a problem (is someone using this id for some reason?)

**Review by aniamusial — APPROVED** [review]

File: packages/vue/src/components/atoms/SfIcon/SfIcon.vue [hunk 1]
```diff
@@ -82,70 +66,53 @@ export default {
...
+    isSFSizes() {
+      const size = this.size.trim();
+      return SF_SIZES.includes(size);
+    },
+    iconColorClass() {
+      return this.isSFColors ? `color-${this.color.trim()}` : "";
+    },
+    iconSizeClass() {
+      return this.isSFSizes ? `size-${this.size.trim()}` : "";
+    },
```
Comment (AdamPawlinski) [inline packages/vue/src/components/atoms/SfIcon/SfIcon.vue:82]: Do you think static classes should be replaced with dynamic ones?

**Review by AdamPawlinski — COMMENTED** [review]
Only one question.

File: packages/vue/src/components/atoms/SfIcon/SfIcon.stories.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ export default {
     docs: {
       description: {
         component:
-          "The icon with color and size modifiers. It's Vue 2 functional component.",
```
Comment (aniamusial) [inline packages/vue/src/components/atoms/SfIcon/SfIcon.stories.js:29]: praise: nice :)

File: packages/vue/src/components/atoms/SfIcon/SfIcon.vue (hunk 1 above)
Comment (aniamusial) [reply packages/vue/src/components/atoms/SfIcon/SfIcon.vue:82]: @AdamPawlinski it's better for tests - with switch we should cover each case of switch. Also, there is a lot of repetitive code with those 'color-' and 'size-' prefixes that can be replaced with single line.

**Review by aniamusial — APPROVED** [review]
great work, it looks way simpler now :)

**Review by AdamPawlinski — APPROVED** [review]

**Review by aniamusial — APPROVED** [review]

**Pull request merged**